app/src/main/java/xyz/whereuat/whereuat/utils/PhonebookUtils.java

```diff
@@ -47,6 +50,33 @@
     }
 
     /**
+     * Look through the phonebook for a contact with phone number |phone| and return their name if
+     * found.
+     *
+     * Note: This function runs a database command and should not be run on the UI thread.
+     *
+     * @param context The Context for the query to use.
+     * @param phone The phone number being searched for.
+     * @return The name of the contact with phone number |phone|.
+     */
+    @NonNull
+    public static String queryPhonebookForContactName(Context context, String phone) {
+        Uri uri = Uri.withAppendedPath(ContactsContract.PhoneLookup.CONTENT_FILTER_URI, phone);
+        Cursor cursor = context.getContentResolver().query(uri,
+                new String[] {ContactsContract.PhoneLookup.DISPLAY_NAME}, null, null, null);
+
+        String name = "";
+
+        if (cursor.moveToFirst()) {
```

Comment:
Is `cursor` guaranteed to be non-null? Could be cleaner to use a try-catch-finally here to handle closing the cursor too